springboottrades-quotes/src/main/java/io/pivotal/quotes/service/QuoteService.java

```diff
@@ -36,165 +25,119 @@
 public class QuoteService {
 	@Value("${pivotal.quotes.quotes_url}")
 	protected String quote_url;
-	@Value("${pivotal.quotes.quotes_url2}")
-	protected String quote_url2;
 	@Value("${pivotal.quotes.companies_url}")
 	protected String company_url;
 
 	@Value("${pivotal.quotes.yahoo_rest_query}")
 	protected String yahoo_url = "https://query.yahooapis.com/v1/public/yql?q=select * from yahoo.finance.quotes where symbol in ('{symbol}')&format={fmt}&env={env}";
 
-	// @Value("${pivotal.quotes.yahoo_query}")
-	// protected String QID =
-	// "select * from yahoo.finance.quotes where symbol in ";
-
 	@Value("${pivotal.quotes.yahoo_env}")
 	protected String ENV = "http://datatables.org/alltables.env";
 
 	public static final String FMT = "json";
 
-	private static final Logger logger = LoggerFactory
-			.getLogger(QuoteService.class);
+	private static final Logger logger = LoggerFactory.getLogger(QuoteService.class);
 
 	/*
 	 * cannot autowire as don't want ribbon here.
 	 */
 	private RestTemplate restTemplate = new RestTemplate();
 
+    /**
+     * Retrieve one or more quotes.
+     *
+     * @param symbols comma delimited list of symbols.
+     * @return a list of quotes.
+     */
+    @HystrixCommand(fallbackMethod = "getMarkitondemandQuotes")
+    public List<Quote> getQuotes(String symbols) throws SymbolNotFoundException {
+        logger.debug("retrieving quotes for: " + symbols);
+        if ( symbols.isEmpty() ) return new ArrayList<>();
+
+        List<YahooQuote> yahooQuotes;
+        Date createDate;
+        if ( symbols.indexOf( ',' ) == -1 ) {
```

Comment:
There is a property to pass to Jackson reader that will read single elements as arrays. no need for tis If statement